keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/SlackService.kt

```diff
@@ -45,4 +46,17 @@ class SlackNotifier(
       log.debug("new slack integration is not enabled")
     }
   }
+
+  fun getUserInfoByEmail(email: String): String? {
+    log.debug("lookup user id for email $email")
+    val response = slack.methods(configToken).usersLookupByEmail { req ->
+      req.email(email)
+    }
+    log.debug("slack returned ${response.isOk}")
+
+    if (response.user != null && response.user.name != null) {
+      return "@${response.user.name}"
+    }
+    return email
```

Comment:
Looks like you're falling back to the email if you can't retrieve the user name, so the return type doesn't have to be nullable.